posts/app.py

```diff
@@ -36,6 +41,18 @@ def upload_new_post():
     Post request files should contain all the files the user wants to upload
     to the server.
     """
+    try:
+        post = {
+            "event_id": request.form["event_id"],
+            "author_id": request.form["author_id"],
+            "text":  request.form["text"],
+            "files": [file for file in request.files.values()]
+        }
+        return str(upload_new_post_to_db(post, DB.posts_collection)), 201
+    except BadRequestKeyError:
```

Comment:
In what case do each of these errors get thrown?

Reply:
`werkzeug.exceptions.BadRequestKeyError` raises when `request.form` can't access an attribute (e.g. when `request.form["event_id"]` fails). The `ValueError` is raised when the user supplies neither a `text` value nor any files since at least one of those is required for a valid post. I just improved the reporting of this in https://github.com/knative-portability/large-events/pull/20/commits/31d1177fffe4cc3e57a42bfc739714f9243fcfd5